app/controllers/application_controller.rb

```diff
@@ -387,6 +388,16 @@ def report_data
     }
   end
 
+  def options_from_session(options)
```

Comment:
Sorry, this is problematic.

I suggest you use the same approach that already works with other nested lists, none of them need to load 7 things from session..

Maybe more generally.. `report_data` should not depend on *anything* in session. It still does in some cases, but, we need to make it less dependent on session if we're ever going to move that to the API.

So.. I suggest passing these arguments in the query string from JS, as with the others, if you *really* need them.

(Except, `where_clause` would be problematic that way, but you don't really need to..)